application/controller/DateController.php

```diff
@@ -1,223 +1,230 @@
-		$years	= array();
-		if($order == "asc"){
-			for($i = $start; $i <= $end; $i++){
-				$years["{$i}"]	= $i;
-			}
-			goto checkDefault;
-		}
-		for($i = $end; $i >= $start; $i--){
-			$years["{$i}"]	= $i;
-		}
-		
-		checkDefault:
-		if(!in_array($default, $years)){
-			goto end;
-		}
-		$years['default']	= $default;
-		
-		end:
-		return $years;
-	}
-	
-	/**
-	 * Checks the full date submitted to see if it is
-	 * valid according to the parameters of the
-	 * Gregorian calander
-	 * 
-	 * @param	int, int, int
-	 * @author	sbebbington
-	 * @date	6 Jul 2017 13:50:32
-	 * @version 0.1.4-RC2
-	 * @return	boolean
-	 */
-	protected function checkDateValidity($day = null, $month = null, $year = null){
-		if($day == null || $month == null || $year == null){
-			return false;
-		}
-		return checkdate($month, $day, $year);
-	}
-	
-	/**
-	 * Returns the default values to the view
-	 * to auto-select day, month, and year
-	 *
-	 * @param	stdClass
-	 * @author	sbebbington
-	 * @date	6 Jul 2017 11:37:21
-	 * @version 0.1.4-RC2
-	 * @return	string | null
-	 */
-	public function getDefault($viewObject = null){
-		if($viewObject == null){
-			return '';
-		}
-		return $viewObject['default'] ?? null;
-	}
-	
-	/**
-	 * Clears the default parameter from the view object
-	 * should one exist
-	 *
-	 * @param	stdClass | array
-	 * @author	sbebbington
-	 * @date	6 Jul 2017 11:49:16
-	 * @version 0.1.4-RC2
-	 * @return	array
-	 */
-	public function clearDefault($viewObject = null){
-		if($viewObject == null){
-			return '';
-		}
-		$viewObject['default']	= null;
-		return array_filter($viewObject, 'strlen');
-	}
+    public function __construct(){
+        ControllerCore::__construct();
+        
+        if(!empty($this->post)){
+            $daySubmitted   = $this->post['day'];
+            $monthSubmitted = $this->post['month'];
+            $yearSubmitted  = $this->post['year'];
+            $dateSubmitted  = "You submitted the following date: {$daySubmitted}/{$monthSubmitted}/{$yearSubmitted} ";
+    
+            if($this->checkDateValidity($daySubmitted, $monthSubmitted, $yearSubmitted) == true){
+                $dateSubmitted  .= " - This is a valid date";
+            }else{
+                $dateSubmitted  .= " - This is not a valid date";
+            }
+            
+            $this->view->dateSubmitted  = $dateSubmitted;
+            goto end;
+        }
+        
+        $day    = array(
+            ''  => "-- Select Day --",
+        );
+        $month    = array(
+            ''  => "-- Select Month --",
+        );
+        $year    = array(
+            ''  => "-- Select Year --",
+        );
+        
+        $day    += $this->setDays((int)date('d'));
+        $month  += $this->setMonths('full', 'numeric', date('m'));
+        $year   += $this->setYears(1901, (int)date('Y'), 'asc', (int)date('Y'));
+        
+        $this->view->days   = $day;
+        $this->view->months = $month;
+        $this->view->years  = $year;    
+        end:
```

Comment:
do we need end: here?

Reply:
Haha - yes, don't you just love it.

On a serious note, now I've met the 'goto' challenge I was set, I'm going to not use it in future revision.